openstack/dws/cluster/requests.go

```diff
@@ -57,6 +84,60 @@ func Delete(c *golangsdk.ServiceClient, id string) (r DeleteResult) {
 			"keep_last_manual_snapshot": 0,
 		},
 	}
+
 	_, r.Err = c.Delete(resourceURL(c, id), reqOpt)
 	return
 }
+
+func List(c *golangsdk.ServiceClient, opts ListOpts) (*ListClustersRst, error) {
+	var rst ListClustersRst
+	q, err := golangsdk.BuildQueryString(opts)
+	if err != nil {
+		return nil, err
+	}
+	url := listURL(c) + q.String()
+
+	_, err = c.Get(url, &rst, &golangsdk.RequestOpts{MoreHeaders: RequestOpts.MoreHeaders})
+	if err == nil {
+		return &rst, nil
+	}
+
+	return nil, err
```

Comment:
maybe `return &rst, err` is better.

Reply:
if exist err, expect to return null rather than an empty obj